### PR TITLE
IRGen: Actually make use of private linkage

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test Linux

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Review by aschwaighofer — APPROVED** [review]
Nice cleanup.

The HiddenNonUnique change should be NFC since we never actually assign that linkage.

The internal linkage change LGTM assuming the debugger does not need access to private symbols (I don't know the history there). @jrose-apple added this code in 2015.

**Comment by slavapestov** [comment]
Yeah, the lldb tests passed so I’m assuming that hack is no longer necessary. But I’ll wait for @rjmccall or @jrose-apple to take a look before merging.

I also noticed we emit value witnesses as linkonce_odr for no apparent reason. I don’t think they need to be ODR except for imported types.

File: include/swift/SIL/FormalLinkage.h
Comment (rjmccall) [inline include/swift/SIL/FormalLinkage.h]: Removing HiddenNonUnique makes sense to me if we never use it currently.  Renaming HiddenUnique seems like it doesn't actually win us anything, and it just creates pain if we ever need to reintroduce HiddenNonUnique.

**Review by rjmccall — COMMENTED** [review]
Otherwise the change looks great.

**Comment by jrose-apple** [comment]
I'm not quite sure what past me was thinking. LLVM `internal` should be fine; it's LLVM `private` that would be a problem. Maybe I got mixed up. Anyway, if the tests are passing then go ahead.

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Pull request merged**
